updater/osv/osv.go

```diff
@@ -554,24 +558,40 @@ func (e *ecs) Insert(ctx context.Context, skipped *stats, name string, a *adviso
 			}
 			// This does some heavy assumptions about valid inputs.
 			ranges := make(url.Values)
-			for _, ev := range r.Events {
+			vs := &rangeVer{rng: &claircore.Range{}}
+			var seenIntroduced bool
+			for ie := range r.Events {
+				ev := r.Events[ie]
 				var err error
 				switch r.Type {
 				case `SEMVER`:
 					var ver *semver.Version
 					switch {
-					case ev.Introduced == "0": // -Inf
-						v.Range.Lower.Kind = `semver`
+					case ev.Introduced != "" && seenIntroduced:
+						vs = &rangeVer{rng: &claircore.Range{}}
```

Comment:
if we are assuming it's well-formed (introduced, fixed, introduced, fixed, etc), then I think we can just do `vs = &rangeVer{rng: &claircore.Range{}}` unconditionally within `case ev.Introduced` and don't need the `if ie == len(r.Events)-1` below. What do you think?

Reply:
I think we need to deal with just (introduced) and (introduced, fixed, introduced) i.e. the cases when the last event is introduced (make sure to append the `vs`), and where it is well terminated (introduced, fixed, introduced, fixed) (leave the `vs` open ended to be potentially filled by another event). The solo introduced case definitely needs to be accounted for, the introduced, fixed, introduced, seems less likely but would also be accounted for.